### PR TITLE
App status for Filtering

### PR DESCRIPTION
**Pull request by kildre**
Added a Server Status for filtering. When user clicks filters they will be able to filter based on the server status.
## Reference Issues or PRs
https://gitlab.jatic.net/jatic/team-metrostar/t-e-platform/-/issues/596

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

#### Text styling

- [x] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [x] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [x] If the content were to be read as plain text, it still makes sense, and no information is missing.

## Any other comments?
image

### CODE REVIEW

File: .husky/pre-push [hunk 1]
```diff
@@ -21,7 +21,7 @@ fi
 
 # Run Unit Tests
 
-npm run test
+# npm run test
 
 # Run Linting
```
Comment (jbouder) [inline .husky/pre-push:27]: Lets remove the test and lint checks entirely from the pre-push check. They don't need to be here.

File: ui/src/pages/home/apps-section/app-filters/app-filters.tsx [hunk 2]
```diff
@@ -209,6 +231,28 @@ export const AppFilters = ({
...
               <Divider sx={{ mt: '24px', mb: '16px' }} />
+              <FormLabel
+                id="server-statuses-label"
+                sx={{
+                  pb: '16px',
+                  fontSize: '14px',
+                  fontWeight: 600,
+                }}
+              >
+                Server Statuses
```
Comment (jbouder) [inline ui/src/pages/home/apps-section/app-filters/app-filters.tsx:242]: This should be `Server Status` according to the design

File: ui/src/utils/jupyterhub.test.ts [hunk 3]
```diff
@@ -184,6 +184,7 @@ describe('JupyterHub utils', () => {
       'all',
       [],
       'Recently modified',
+      [], // Add an empty array as the seventh argument
```
Comment (jbouder) [inline ui/src/utils/jupyterhub.test.ts:187]: This comment doesn't seem to add any value, I suggest removing it

File: ui/src/utils/jupyterhub.test.ts [hunk 4]
```diff
@@ -196,6 +197,7 @@ describe('JupyterHub utils', () => {
       'all',
       [],
       'Name: A-Z',
+      [], // Add an empty array as the seventh argument
```
Comment (jbouder) [inline ui/src/utils/jupyterhub.test.ts:200]: Same ^^

File: ui/src/utils/jupyterhub.test.ts [hunk 5]
```diff
@@ -208,6 +210,7 @@ describe('JupyterHub utils', () => {
       'all',
       [],
       'Name: Z-A',
+      [], // Add an empty array as the seventh argument
```
Comment (jbouder) [inline ui/src/utils/jupyterhub.test.ts:213]: Same ^^

**Review by jbouder — CHANGES_REQUESTED** [review]
In addition to the comments, lets also update the jest config as discussed to run tests on 1 worker, setting like this: `maxWorkers: 1`, in order to prevent intermittent test failures.

File: ui/src/utils/jupyterhub.test.ts (hunk 3 above)
Comment (kildre) [reply ui/src/utils/jupyterhub.test.ts:187]: Removed

File: ui/src/pages/home/apps-section/app-filters/app-filters.tsx (hunk 2 above)
Comment (kildre) [reply ui/src/pages/home/apps-section/app-filters/app-filters.tsx:242]: Updated

File: .husky/pre-push (hunk 1 above)
Comment (kildre) [reply .husky/pre-push:27]: Uncommented

File: ui/src/utils/jupyterhub.test.ts (hunk 4 above)
Comment (kildre) [reply ui/src/utils/jupyterhub.test.ts:200]: Removed

File: ui/src/utils/jupyterhub.test.ts (hunk 5 above)
Comment (kildre) [reply ui/src/utils/jupyterhub.test.ts:213]: Removed

**Comment by kildre** [comment]
> In addition to the comments, lets also update the jest config as discussed to run tests on 1 worker, setting like this: `maxWorkers: 1`, in order to prevent intermittent test failures.

Updated

**Review by jbouder — APPROVED** [review]
Looks great!

**Pull request merged**